### ISSUE TITLE
`aria-describedby` attribute is set incorrectly

### ISSUE THREAD

**Issue by webmandesign** [woothemes/woocommerce#38801]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

Since WooCommerce 7.8.0, the `aria-describedby` attribute set on the add to cart button of variable products in products list renders an accessibility error (tested with [WAVE](https://wave.webaim.org/)).

### Expected behavior

[`aria-describedby` attribute](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-describedby) value should be set to an ID of an HTML element containing the description text.

(To fix the issue, perhaps [`aria-description` attribute](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-description) can be used instead, depending on the intention of this WooCommerce 7.8.0 code.)

### Actual behavior

In WooCommerce 7.8.0 the [value of `aria-describedby` attribute](https://github.com/woocommerce/woocommerce/blob/7.8.0/plugins/woocommerce/includes/class-wc-product-variable.php#L65) is set with the actual description text, which is incorrect.

### Steps to reproduce

1. Take a WooCommerce 7.8.0 website
2. Navigate to a page containing a list of products where at least one product is a variable product.
3. Test the page with https://wave.webaim.org/ and check for accessibility errors.

Example: https://wave.webaim.org/report#/https://themedemos.webmandesign.eu/eimear/product-category/furniture/

### WordPress Environment

n/a

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by umarsear** [comment]
I am seeing similar issue since upgrading; is there something I can do to help in reproducing the issue so that it can be resolved?

**Comment by shameemreza** [comment]
Another report here: https://wordpress.org/support/topic/wave-aria-labelledby-or-aria-describedby-issue-on-variable-products/

7357648-zen

**Comment by Mldiguardi** [comment]
With the amount of lawsuits against small businesses continuing to be on the rise in the U.S. for ADA Compliance websites the "Broken ARIA" Reference is an issue.  The broken ARIA reference is on all variable product where the "Select Options" appears (woocommerce category pages, product pages that have linked products, etc). 

I had found the [below code here](https://sridharkatakam.com/remove-select-options-button-variable-products-main-woocommerce-shop-page/) which removed the "Select Options" on the variable products, which then corrected the "Broken ARIA" Reference. 

The display of the "Select Options" on ecommerce, not only looks better but ALSO lets a visiting customer know that there are additional choices of the product(s). 

```
// Remove "Select options" button from (variable) products on the main WooCommerce shop page.
add_filter( 'woocommerce_loop_add_to_cart_link', function( $product ) {

	global $product;

	if ( is_shop() && 'variable' === $product->product_type ) {
		return '';
	} else {
		sprintf( '<a href="%s" data-quantity="%s" class="%s" %s>%s</a>',
			esc_url( $product->add_to_cart_url() ),
			esc_attr( isset( $args['quantity'] ) ? $args['quantity'] : 1 ),
			esc_attr( isset( $args['class'] ) ? $args['class'] : 'button' ),
			isset( $args['attributes'] ) ? wc_implode_html_attributes( $args['attributes'] ) : '',
			esc_html( $product->add_to_cart_text() )
		);
	}

} );
```

Thank you.

**Comment by ju1ius** [comment]
> status: needs reproduction

This bug does not need reproduction. It is reproducable on _any_ woocommerce installation.

The [woocommerce_template_loop_add_to_cart](https://github.com/woocommerce/woocommerce/blob/f9c4b284ce11aba542ce7a03814ca8e96d0b6d5e/plugins/woocommerce/includes/wc-template-functions.php#L1385) function adds the `aria-describedby` attribute by calling  [WC_Product::add_to_cart_aria_describedby()](https://github.com/woocommerce/woocommerce/blob/f9c4b284ce11aba542ce7a03814ca8e96d0b6d5e/plugins/woocommerce/includes/abstracts/abstract-wc-product.php#L1944), which is documented as such:

```php
/**
 * Get the aria-describedby description for the add to cart button.
 *
 * @return string
 */
public function add_to_cart_aria_describedby() {}
```

We can clearly see there's a fundamental misunderstanding here. The `aria-describedby` should contain _the ID of a DOM element containing the description_, not the description itself.
